ydb/library/yql/dq/opt/dq_opt_conflict_rules_collector.cpp

```diff
@@ -0,0 +1,86 @@
+#include "dq_opt_conflict_rules_collector.h"
+#include <util/generic/hash_set.h>
+
+namespace NYql::NDq {
+
+/* To make ASSOC, RASSCOM, LASSCOM tables simplier */
+EJoinKind GetEquivalentByAlgebraicPropertiesJoin(EJoinKind joinKind) {
+    switch (joinKind) {
+        case EJoinKind::Exclusion:
+            return EJoinKind::InnerJoin;
+        case EJoinKind::LeftOnly:
+            return EJoinKind::LeftJoin;
+        default:
+            return joinKind;
+    }    
+}
+
+bool OperatorIsCommut(EJoinKind joinKind) {
+    joinKind = GetEquivalentByAlgebraicPropertiesJoin(joinKind);
+    switch (joinKind) {
+        case EJoinKind::InnerJoin:
+        case EJoinKind::OuterJoin:
+        case EJoinKind::Cross:
+            return true;
+        default:
+            return false;
+    }
+
+    Y_UNREACHABLE();
+}
+
+bool OperatorsAreAssoc(EJoinKind lhs, EJoinKind rhs) {
+    lhs = GetEquivalentByAlgebraicPropertiesJoin(lhs);
+    rhs = GetEquivalentByAlgebraicPropertiesJoin(rhs);
+
+    static THashMap<EJoinKind, THashSet<EJoinKind>> ASSOC_TABLE = {
+        {EJoinKind::Cross, {EJoinKind::Cross, EJoinKind::InnerJoin, EJoinKind::LeftSemi, EJoinKind::LeftJoin}},
+        {EJoinKind::InnerJoin, {EJoinKind::Cross, EJoinKind::InnerJoin, EJoinKind::LeftSemi, EJoinKind::LeftJoin}},
+        {EJoinKind::LeftJoin, {EJoinKind::LeftJoin}},
+        {EJoinKind::OuterJoin, {EJoinKind::LeftJoin, EJoinKind::OuterJoin}}
+    };
+
+    if (!(ASSOC_TABLE.contains(lhs))) {
+        return false;
+    }
+
+    return ASSOC_TABLE[lhs].contains(rhs);
+}
+
+bool OperatorsAreLeftAsscom(EJoinKind lhs, EJoinKind rhs) {
```

Comment:
Write a short header specifying what LeftAsscom means, or just put in a reference to the paper